pkg/dataobj/planner/logical/aggregate_expr.go

```diff
@@ -0,0 +1,109 @@
+package logical
+
+import (
+	"github.com/grafana/loki/v3/pkg/dataobj/internal/metadata/datasetmd"
+	"github.com/grafana/loki/v3/pkg/dataobj/planner/schema"
+)
+
+// AggregateOp represents the type of aggregation operation to perform.
+// It is a string-based enum that identifies different aggregation functions
+// that can be applied to expressions.
+type AggregateOp string
+
+const (
+	// AggregateOpSum represents a sum aggregation
+	AggregateOpSum AggregateOp = "sum"
+	// AggregateOpAvg represents an average aggregation
+	AggregateOpAvg AggregateOp = "avg"
+	// AggregateOpMin represents a minimum value aggregation
+	AggregateOpMin AggregateOp = "min"
+	// AggregateOpMax represents a maximum value aggregation
+	AggregateOpMax AggregateOp = "max"
+	// AggregateOpCount represents a count aggregation
+	AggregateOpCount AggregateOp = "count"
+)
```

Comment:
I think we should use the same enum pattern (int type + iota + String() function) as in the rest of the PR.